packages/docs/src/components/DocSiteNavigation.tsx

```diff
@@ -0,0 +1,221 @@
+      allFile(
+        sort: { fields: [relativeDirectory, name] }
+        filter: { ext: { eq: ".mdx" }, relativeDirectory: { ne: "not-in-sidebar" } }
+      ) {
+        group(field: relativeDirectory) {
+          fieldValue
+          edges {
+            node {
+              id
+              name
+              relativeDirectory
+              relativePath
+            }
+          }
+        }
+      }
+    }
+  `);
+
+  /**
+   * determines if the item name is included in the location pathname
+   */
+  const isItemSelected = (name: string) => {
+    return location?.pathname.includes(name);
+  };
+
+  /**
+   * Checks sub nav items to see if any are currently selected
+   */
+  const isSubNavItemSelected = (subNavItems) => {
+    return subNavItems.some((navItem) => isItemSelected(navItem.url));
+  };
+
+  /**
+   * Updating a name to remove kebab case & get rid of numeric ordering
+   */
+  const formatNavItemLabel = (name: string): string => {
+    let newName = name.replace(/-/g, ' ');
+    newName = removePositioning(newName);
+    return newName;
+  };
+
+  const formatNavItemData = ({ name, relativePath }: NavItem) => {
+    const url = makePageUrl(relativePath);
+    return {
+      label: formatNavItemLabel(name),
+      url,
+      id: url,
+      selected: isItemSelected(url),
+    };
+  };
+
+  const formatNavData = (dataList: GraphQlNavItem[]): VerticalNavItemProps[] => {
+    const retVal: VerticalNavItemProps[] = [];
+    dataList.forEach((dataItem) => {
+      // for each level 1 item that has sub nav items
+      if (dataItem.fieldValue.length) {
+        // format all the level 2 items
+        const subNavItems = dataItem.edges.map((subNavItem) => formatNavItemData(subNavItem.node));
+
+        const labelText = formatNavItemLabel(dataItem.fieldValue);
+        const isSelected = isItemSelected(dataItem.fieldValue);
+        // add level 1 item & sub items
+        retVal.push({
+          label: labelText,
+          items: subNavItems,
+          defaultCollapsed: !isSubNavItemSelected(subNavItems),
+          selected: isSelected,
+        });
+      } else {
+        // for each level 1 item without sub nav items,
+        // add each top level item
+        dataItem.edges.forEach((navItemLvl1) => {
+          retVal.push(formatNavItemData(navItemLvl1.node));
+        });
+      }
+    });
+
+    return retVal;
+  };
+
+  const navItems: VerticalNavItemProps[] = formatNavData(data?.allFile?.group);
+
+  return (
+    <div
+      className={classnames('ds-u-padding--0 ds-u-md-padding--2 c-navigation', {
+        'c-navigation--open': isMobile && isMobileNavOpen,
+      })}
+    >
+      <header className="c-navigation__header">
+        <Button
+          className="ds-u-md-display--none ds-u-padding-left--0 ds-u-padding-right--1"
+          variation="transparent"
+          aria-expanded={isMobileNavOpen}
+          aria-controls="c-mobile-navigation"
+          onClick={toggleMenu}
+        >
```

Comment:
One tiny thing that I think could help make this look really polished is to normalize the gap around this focus ring:
![menu-button-focus](https://user-images.githubusercontent.com/7595652/175999865-b24fddb7-ec4b-47d9-bc14-45b3f9785c20.gif)

Reply:
By "normalize," do you mean center position the icon within the button? 

My one concern with that is if the button doesn't have focus, the icon will look off-kilter with the container width on the page. I kept the right padding in place to give users more click affordance, but maybe it's best to reduce it to the same width as the left-padding to visually center it? 

@line47 do you have a preference on this?

Reply:
I didn't have a plan completely thought out, but I figured you could do something with negative margin to get the button to appear in the right place but still have even spacing around the icon. (Yes, I mean centering the icon)